components/gripper/trossen/trossen.go

```diff
@@ -81,162 +78,58 @@ func init() {
+	var _arm *trossenarm.Arm
+	a, err := arm.FromDependencies(deps, attr.Arm)
 	if err != nil {
 		return nil, err
 	}
-	if err := jServo.SetTorqueEnable(true); err != nil {
-		return nil, err
-	}
-	newGripper := Gripper{
-		jServo:   jServo,
-		moveLock: getPortMutex(usbPort),
+	_arm, ok := rdkutils.UnwrapProxy(a).(*trossenarm.Arm)
+	if !ok {
+		return nil, errors.Errorf(
+			"arm specified for trossen gripper %s is not a trossen arm", name)
 	}
+	newGripper := Gripper{trossenArm: _arm, logger: logger, name: name}
 	return &newGripper, nil
 }
 
-// GetMoveLock TODO.
-func (g *Gripper) GetMoveLock() *sync.Mutex {
-	return g.moveLock
-}
-
-// Open TODO.
+// Open opens the gripper by defering to the arm.
 func (g *Gripper) Open(ctx context.Context, extra map[string]interface{}) error {
-	ctx, done := g.opMgr.New(ctx)
-	defer done()
-	g.moveLock.Lock()
-	defer g.moveLock.Unlock()
-	err := g.jServo.SetGoalPWM(150)
-	if err != nil {
-		return err
+	if err := g.trossenArm.OpenGripper(ctx); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("open failed for trossen gripper %s", g.name))
 	}
-
-	// We don't want to over-open
-	atPos := false
-	for !atPos {
-		var pos int
-		pos, err = g.jServo.PresentPosition()
-		if err != nil {
-			return err
-		}
-		if pos < 2800 {
-			if !utils.SelectContextOrWait(ctx, 50*time.Millisecond) {
-				return ctx.Err()
-			}
-		} else {
-			atPos = true
-		}
-	}
-	err = g.jServo.SetGoalPWM(0)
-	return err
+	return nil
 }
 
-// Grab TODO.
+// Grab closes the gripper by defering to the arm.
 func (g *Gripper) Grab(ctx context.Context, extra map[string]interface{}) (bool, error) {
-	_, done := g.opMgr.New(ctx)
-	defer done()
-	g.moveLock.Lock()
-	defer g.moveLock.Unlock()
-	err := g.jServo.SetGoalPWM(-350)
+	hasGrabbed, err := g.trossenArm.Grab(ctx)
 	if err != nil {
-		return false, err
+		return false, errors.Wrap(err, fmt.Sprintf("open failed for trossen gripper %s", g.name))
 	}
-	err = servo.WaitForMovementVar(g.jServo)
-	if err != nil {
-		return false, err
-	}
-	pos, err := g.jServo.PresentPosition()
-	if err != nil {
-		return false, err
-	}
-	didGrab := true
-
-	// If servo position is less than 1500, it's closed and we grabbed nothing
-	if pos < 1500 {
-		didGrab = false
-	}
-	return didGrab, nil
+	return hasGrabbed, nil
 }
 
-// Stop is unimplemented for Gripper.
+// Stop is stops the gripper servo on the trossen arm.
 func (g *Gripper) Stop(ctx context.Context, extra map[string]interface{}) error {
-	return multierr.Combine(
-		g.jServo.SetTorqueEnable(false),
-		g.jServo.SetTorqueEnable(true),
-	)
+	return g.trossenArm.Stop(ctx, extra)
 }
 
 // IsMoving returns whether the gripper is moving.
 func (g *Gripper) IsMoving(ctx context.Context) (bool, error) {
-	return g.opMgr.OpRunning(), nil
-}
-
-// Close closes the connection, not the gripper.
-func (g *Gripper) Close() error {
-	return g.jServo.SetTorqueEnable(false)
+	return g.trossenArm.IsMoving(ctx)
```

Comment:
[nit] same here as for stop, do we want to just have this handle gripper-specific movement?